docker/docker.go

```diff
@@ -519,3 +519,52 @@ func GetLogs(ctx context.Context, containerID string) (io.ReadCloser, error) {
 
 	return reader, err
 }
+
+func Attach(ctx context.Context, config *container.Config, host *container.HostConfig, name string) (*types.HijackedResponse, chan int64, error) {
```

Comment:
By `Attach` it sounds like this command should be used like `docker attach`, over a running container. So you call `Start` and then `Attach`.

But this method already starts the container. 
I guess you need to it this way for the attach config settings. Is there a way to attach to a running container? I think it would be more clear.
It not, I would consider changing the method name. `StartAttach`? And documenting what it does.

Reply:
this command actually works as `docker run -it`. The difference from normal `docker attach` is that it attaches BEFORE starting the container. So you have all the output of the container.
You can try with docker by running `docker run mysql mysql -e "select * from files"` and then `docker attach`. You will get only part of the output since the moment you attached.

I don't see any use case when engine would need attach to a running container. So I named it just `Attach` not `AttachRun` or `StartAttach`. But I don't mind to rename it if you prefer so.

Later we can implement `exec` that actually attaches to a running container and runs a command. It should be very similar code. (most probably reuse parts of this function). Though for `exec` we need image that runs something as pid 1 and doesn't exit. Like new gitbase. For the current image `run` is a better fit.

Reply:
Thanks for the explanation. Attaching before starting the container is a very good reason to keep it like it's now :+1:.

Maybe we could add comment explaining that it's similar to `run -it`

Reply:
done